### ISSUE TITLE
Unable to Build for ARM

### ISSUE THREAD

**Issue by riceke** [Team-Hycon/hycon-core#7]
Hello!

I am trying to build for linux on Armbian OS (Debian based) and I get a gcc compiler error (for target 'Release/obj.target/cryptonight/lib/cryptonight.o') because the compiler command line option '-maes' is not present for ARM platforms.

Since I am not planning to do any mining but just running an always-on node I don't think that it matters that much that there is no AES hardware acceleration available, or?

Is there a way to disable the usage of the '-maes' flag when comiling for ARM?

Thanks!

**Comment by ChosunOne** [comment]
You can get it to build if you use my fork of cryptonight instead of the official one in your package.json.  Basically you can just remove the "maes" flags from the build options and it ought to be ok (at least that is how i got my pi running with it).

**Comment by Team-Hycon** [comment]
Was this resolved?

**Comment by riceke** [comment]
Hello!

Yes, I managed to build on my ASUS Tinkerboard (which is a 32 bit ARM architecture) by cloning the 'node-cryptonight' repository and modifying the 'binding.gyp' to replace the '-maes' flag with '-marm'.

I had some issues though with the fact that the 'node-modules' directory under 'platform/linux' did not seem to be updated with the locally built modules when running 'package.sh', so the node modules in the 'bundle-linux' directory ended up being the default AMD64 ones instead of the locally built ARM32 ones. Once I figured out what was going on I just replaced the node module files in the 'platform/linux/node-modules' directory with some symbolic links to the locally built modules.

Thanks!
